src/spikeinterface/postprocessing/amplitude_scalings.py

```diff
@@ -415,33 +445,43 @@ def find_collisions(spikes, spikes_w_margin, delta_collision_samples, sparsity_m
     # TODO: refactor to speed-up
     collision_spikes_dict = {}
     for spike_index, spike in enumerate(spikes):
-        # find the index of the spike within the spikes_w_margin
-        spike_index_w_margin = np.where(spikes_w_margin == spike)[0][0]
 
-        # find the possible spikes per and post within delta_collision_samples
+        # find the index of the spike within spikes_within_margin
+        spike_index_within_margin = np.where(spikes_within_margin == spike)[0][0]
```

Comment:
I think you are right. My coment was that I like the semantics of `np.where` a bit more. As in, it is more clear what it is doing. But now that I think about it as I write this, maybe not so clear x D

Reply:
Cool, I did not know about nonzero! I see indeed numpy does [recommend its use](https://numpy.org/doc/stable/reference/generated/numpy.where.html) when the additional arguments are not provided. I don't quite understand the reason, they say `Using nonzero directly should be preferred, as it behaves correctly for subclasses.` but I'm not sure what that means - if `np.where` is just wrapping `np.nonzero` in this case then I wonder why it is performing differently? 

Personally I prefer `np.where` because:
1) It reads like a normal sentence "spike_index_within_margin equals where 'spikes_within_margin' equals  'spike'"
2) If mixing `np.nonzero` and `np.where` there are now two functions in use for doing essentially the same job.

Maybe we can make an issue on this or have some discussion it on a contributing / a style guide? If it's okay with everyone I will leave it as-is in this PR as it wasn't a change I introduced specifically but the additional was from refactoring some already-existing code.

Reply:
Yeah barring some major issue I think it's fine. An issue to discuss this sounds fine.

Reply:
Given that the three of us prefer np.where instead of np.nonzero and there is not a performance cost maybe let's just adapta that convention. I honestly don't think that Sam or Alessio care about these things : )

![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/9073e723-6a99-430e-a0ff-7f0ff1b1a054)

Reply:
Nice! that sounds good to me, I will open a PR to make some additions to the style guide